### PR TITLE
Add doc for server requirements

### PR DESCRIPTION
**Pull request by xuebinsu**
Add doc for server requirements

As more (experimental) new features are added, it is getting more and
more complex to setup a server to meet the requirements.

To make the process easier, this patch adds a page `requirements.rst`,
to specify the requirements for client and for server. It is for all
the features including the experiemntal ones.

This patch also makes minor adjustments to make the interface of the
experimental features more clear and easier to work with.

### CODE REVIEW

**Review by yihong0618 — APPROVED** [review]

File: doc/source/requirements.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+We also try our best to support PostgreSQL and other PostgreSQL-derived databases, but some features might **NOT** be available when working with them.
+
+.. _Getting Started:
+
+Getting Started
+^^^^^^^^^^^^^^^
+
+To get started, all we need is a database that we have the permission to access.
+
+After connecting to the database, we can create :class:`~dataframe.DataFrame` 's and manipulate them like using `pandas <https://pandas.pydata.org/>`_.
```
Comment (ruxuez) [inline doc/source/requirements.rst:35]: :class:`~dataframe.DataFrame`s without '?

**Review by ruxuez — APPROVED** [review]
Very nice doc!

File: doc/source/notebooks/package.ipynb [hunk 2]
```diff
@@ -18,26 +18,28 @@
...
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## (Optional) Prerequisite: Setting-up an NFS Mount for Cluster\n",
+    "## (Optional) Prerequisite: Sharing Python Environments in a Cluster with NFS\n",
     "\n",
     "Setting up a NFS mount makes it easier to share a Python environment on multiple hosts and containers.\n",
     "\n",
-    "This is important for distributed database systems such as [Greenplum](https://greenplum.org/) because otherwise the same set of pcakges need to be installed on every host in the cluster.\n",
+    "This is important for distributed database systems such as [Greenplum](https://greenplum.org/) because otherwise the same set of pcakges need to be copied to every host in the cluster.\n",
```
Comment (beeender) [inline doc/source/notebooks/package.ipynb:34]: `pcakges` typo

File: doc/source/requirements.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+    CREATE EXTENSION plpython3u;
+
+There are a few points to note when working with `PL/Python`:
+
+- To use the extension :code:`plpython3u`, it is required to login as a :code:`SUPERUSER`. This might cause some security concerns. 
+  We will remove this limitation soon by supporting `PL/Container <https://github.com/greenplum-db/plcontainer>`_.
+- Python 3.x is required on server. Please make sure that :code:`libpython3.x.so` is in :code:`$LD_LIBRARY_PATH`.
+  
+  There is no hard requiremet on the minor version, but Python language features and libraries available for use in a UDF depends on it.
```
Comment (beeender) [inline doc/source/requirements.rst:56]: `requiremet` typo

File: doc/source/requirements.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+- the `Getting Started`_ section,
+- the `Creating Functions`_ section,
+
+plus the `sentence-transformers <https://pypi.org/project/sentence-transformers/>`_ package installed
+in the server's Python environment.
+
+Please refer to the :doc:`tutorial <./tutorial_embedding>` for a simple working example to validate your setup.
+
+Uplading Data Files from Localhost (Experimental)
```
Comment (beeender) [inline doc/source/requirements.rst:93]: `Uplading` typo

**Review by beeender — APPROVED** [review]

**Pull request merged**